### PR TITLE
Solve day2b

### PR DESCRIPTION
**Pull request by haynaou**
@shadanan I managed to get a ⭐️
Can you please review and merge. Thanks 😊

### CODE REVIEW

File: day2b.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        count = int(count)
+        if max_colors[color] < count:
+            max_colors[color] = count
+    
+    # Compute the product of all values
+    power = 1    
+    for value in max_colors.values():
+        power *= value
+    
+    sum_power += power
```
Comment (shadanan) [inline day2b.py]: Did you know there's a `prod` function for this? First you `from math import prod`, then:
```suggestion
    # Compute the sum of product of all values
    sum_power += prod(max_colors.values())
```

File: day2b.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+with open("input2a.txt", "r") as fp:
+    lines = fp.read().strip().splitlines()
+
+sum_power = 0
+for line in lines:
+    game_str, remainder = line.split(": ")
+    game_id = int(game_str.split()[-1])
+    game = remainder.replace("; ", ", ").split(", ")
+    
+    max_colors = {'red': 0, 'green': 0, 'blue': 0}
```
Comment (shadanan) [inline day2b.py]: If we want to be generic to be able to handle any arbitrary colors, we can do that with a `defaultdict`. First, `from collections import defaultdict`, then:
```suggestion
    max_colors = defaultdict(int)
```

**Review by shadanan — APPROVED** [review]

File: day2b.py (hunk 2 above)
Comment (haynaou) [reply day2b.py]: Pretty neat! thanks

**Comment by shadanan** [comment]
BTW, @haynaou - the changes I suggested do require some additional modifications to the file that were not contained in my suggestion – namely, you have to add the imports to the top of the file. As is, these scripts won't work!

**Pull request merged**